src/components/permissions/routes.ts

```diff
@@ -9,17 +12,28 @@ const routeUserPermissions = async (request: any, reply: any) => {
 }
 
 const routeLogin = async (request: any, reply: any) => {
-  const { username, passwordHash } = request.body || { username: '', passwordHash: '' }
+  const { username, password } = request.body || { username: '', password: '' }
+  const { passwordHash } = (await databaseConnect.getUserDataByUsername(username)) || {}
 
-  if (!(await databaseConnect.verifyUser(username, passwordHash)))
-    return reply.send({ success: false })
+  if (!passwordHash) return reply.send({ success: false })
 
-  return reply.send({
+  if (!(await bcrypt.compare(password, passwordHash))) return reply.send({ success: false })
```

Comment:
I am a little bit confused by this. Pardon my ignorance/misunderstanding, but I thought:

Salt is prefixed/postfixed to password before hashing for extra protection. So how could bcrypt.compare validate raw password vs it's salted hash, without knowing the hash ?

Reply:
Ok never mind, it looks like bcrypt output incorporates both salt and hash

Reply:
Yes, I was confused about that initially too. But yeah, the hash string is actually a concatenation of the random salt and the computed hash, as explained [in the docs](https://github.com/kelektiv/node.bcrypt.js#hash-info).

```
$2b$10$nOUIs5kJ7naTuTFkBy1veuK0kSxUFXfuaOKdOKf9xYT0KKIGSJwFa
 |  |  |                     |
 |  |  |                     hash-value = K0kSxUFXfuaOKdOKf9xYT0KKIGSJwFa
 |  |  |
 |  |  salt = nOUIs5kJ7naTuTFkBy1veu
 |  |
 |  cost-factor => 10 = 2^10 rounds
 |
 hash-algorithm identifier => 2b = BCrypt
```

Pretty nifty. :)